pilot/pkg/xds/discovery.go

```diff
@@ -164,9 +166,10 @@ func NewDiscoveryServer(env *model.Environment, instanceID string, clusterID clu
 			debounceMax:       features.DebounceMax,
 			enableEDSDebounce: features.EnableEDSDebounce,
 		},
-		Cache:      model.DisabledCache{},
-		instanceID: instanceID,
-		clusterID:  clusterID,
+		Cache:            model.DisabledCache{},
+		instanceID:       instanceID,
+		clusterID:        clusterID,
+		processStartTime: time.Now(),
```

Comment:
I'm not sure if this is the best place where to initialise this variable.. that's the closest scenario to the original one, but maybe in the `Start` method makes more sense?

Reply:
A bit worried about this change. its not really the processStartTime if we do it anywhere other than init() (technically that is wrong too -- but the best we can do afaik).

Do we need to set it in fake.go at all or can we just remove that? I guess it was probably to make the test logs not look weird.

Maybe we keep processStartTime but make a variable like `discoveryStartTime: processStartTime`, and in fake.go we override that?

Reply:
thank you for your input, indeed I was wrong, the data race was only because of `fake.go` setting `processStartTime`, no need to change the structure of `DiscoveryService`.

If I understood correctly, that should address your concerns.

Reply:
I was thinking we could keep the change you made, but instead have it default to the global `processStartTime` instead of time.Now.

Then in real world, its set to the global. But in fake, we can still reset the per-fake copy?

Otherwise the test may log "synced after 10min" which is confusing. Not a big deal though

Reply:
So, if I understand you correctly, we would be adding a `discoveryStartTime` to `DiscoveryService`, and initialize it to the global `processStartTime`, while `fake.go` would be setting the `discoveryStartTime` directly.

But, if I may ask, what is measuring `processStartTime`? It's used to output the time needed to sync the informer, but.. shouldn't this time start when the informers start to sync?  What am I missing?

Reply:
What I was thinking is

For real binary: we want to know things relative to the time the binary started

For tests: the binary start time is not really relevant. We would want to know the time since the _test_ started, but we cannot really figure that outside, so we can approximate it by setting/overriding it in fake.go to time.Now

Reply:
ops sorry, I didn't have time to address this before it got merged.. do you want me to submit another PR?